frontend/src/metabase-lib/expressions/process.js

```diff
@@ -31,13 +33,23 @@ export function processSource(options) {
     }
   };
 
-  const { source, startRule } = options;
+  const { source, query, stageIndex, startRule } = options;
 
-  let expression;
+  let expression = null;
+  let expressionClause = null;
   let compileError;
   try {
     const parsed = parse(source);
     expression = adjustBooleans(resolve(parsed, startRule, resolveMBQLField));
+
+    // query and stageIndex are not available outside of notebook editor (e.g. in Metrics or Segments).
+    if (typeof query !== "undefined" && typeof stageIndex !== "undefined") {
```

Comment:
`query` seems to be an object, so just `if (query &&` should be enough

Reply:
e6a0bed8dac2a2550be912aba500b99887c8a0a9